src/main/java/org/codehaus/groovy/runtime/GStringImpl.java

```diff
@@ -351,15 +353,26 @@ public String toString() {
         return str;
     }
 
-    private static boolean checkValuesImmutable(Object[] values) {
+    private static boolean checkValuesStringConstant(Object[] values) {
         for (Object value : values) {
             if (null == value) continue;
-            if (!(ImmutablePropertyUtils.builtinOrMarkedImmutableClass(value.getClass())
+            if (!(toStringMarkedPure(value.getClass())
```

Comment:
We can probably swap the order of this clause and the next since this one is probably less common than next line.